theme/management/commands/report_resource_size_vs_quota_inconsistency.py

```diff
@@ -0,0 +1,162 @@
+from hs_core.hydroshare import current_site_url
+from hs_core.models import ResourceFile
+
+current_site = current_site_url()
+_BATCH_SIZE = settings.BULK_UPDATE_CREATE_BATCH_SIZE
+
+
+def get_dz_quota_usage_from_irods(username):
+    """
+    Query iRODS AVU to get quota usage for a user reported in iRODS quota microservices
+    :param username: the user name to get quota usage for.
+    :return: the quota usage from iRODS data zone
+    :raises: ValidationError if quota usage cannot be retrieved from iRODS
+    """
+    attname = username + '-usage'
+    istorage = IrodsStorage()
+    # get quota size for user in iRODS data zone by retrieving AVU set on irods bagit path
+    # collection
+    try:
+        uqDataZoneSize = istorage.getAVU(settings.IRODS_BAGIT_PATH, attname)
+        if uqDataZoneSize is None:
+            # user may not have resources in data zone, so corresponding quota size AVU may not
+            # exist for this user
+            uqDataZoneSize = -1
+        else:
+            uqDataZoneSize = float(uqDataZoneSize)
+    except SessionException:
+        # user may not have resources in data zone, so corresponding quota size AVU may not exist
+        # for this user
+        uqDataZoneSize = -1
+
+    if uqDataZoneSize < 0:
+        err_msg = 'no quota size AVU in data zone and user zone for user {}'.format(username)
+        print(err_msg)
+        raise ValidationError(err_msg)
+    else:
+        used_val = uqDataZoneSize
+    return used_val
+
+
+class Command(BaseCommand):
+    help = "Output potential quota inconsistencies between iRODS and Django resource aggregate filesize for all users"
+
+    def add_arguments(self, parser):
+        parser.add_argument('output_file_name_with_path', help='output file name with path')
+        parser.add_argument('--update', action='store_true', help='fix inconsistencies by recalculating in django')
+        parser.add_argument('--reset', action='store_true', help='reset resource file size in django when inconsistent')
+        parser.add_argument('--uid', help='filter to just a single user by uid')
+
+    def handle(self, *args, **options):
+        quota_report_list = []
+        uid = options['uid'] if options['uid'] else None
+        update = options['update']
+        reset = options['reset']
+
+        if update and reset:
+            print('Cannot use both --update and --reset options together')
+            return
+        uqs = UserQuota.objects.filter(user__is_active=True).filter(user__is_superuser=False)
+        if uid:
+            uqs = uqs.filter(user__id=uid)
+        for uq in uqs:
+            user = uq.user
+            print("\n" + "*" * 80)
+            print(f'Checking quota for user {user.username}, {current_site}/user/{user.id}/')
+            used_value_irods_dz = 0.0
+            try:
+                used_value_irods_dz = get_dz_quota_usage_from_irods(user.username)
+            except ValidationError:
+                pass
```

Comment:
You probably want `continue` here

Reply:
I want to keep checking, using default `used_value_irods_dz = 0.0`